server/service/orbit_client.go

```diff
@@ -140,22 +145,28 @@ func (oc *OrbitClient) GetConfig() (*fleet.OrbitConfig, error) {
 	// If time-to-live passed, we update the config cache
 	now := time.Now()
 	if now.After(oc.configCache.lastUpdated.Add(configCacheTTL)) {
+		verb, path := "POST", "/api/fleet/orbit/config"
 		var (
 			resp fleet.OrbitConfig
 			err  error
 		)
-		verb, path := "POST", "/api/fleet/orbit/config"
-		// Retry until we don't get a network error.
+		// Retry until we don't get a network error or a 5XX error.
 		_ = retry.Do(func() error {
 			err = oc.authenticatedRequest(verb, path, &orbitGetConfigRequest{}, &resp)
-			var netErr net.Error
-			if errors.As(err, &netErr) {
+			var (
+				netErr        net.Error
+				statusCodeErr *statusCodeErr
+			)
+			if err != nil && oc.onGetConfigErrFns != nil && oc.onGetConfigErrFns.DebugErrFunc != nil {
+				oc.onGetConfigErrFns.DebugErrFunc(err)
+			}
+			if errors.As(err, &netErr) || (errors.As(err, &statusCodeErr) && statusCodeErr.code >= 500) {
```

Comment:
Some infrastructure may use custom error codes above 599, but I think it is OK to include them.